### PR TITLE
minor: refactor fadeIn code

### PR DESCRIPTION
**Pull request by RomanZavodskikh**
Some minor changes after https://github.com/zalando/skipper/pull/2634 just to make code a little bit more readable.

### CODE REVIEW

File: proxy/fadein.go [hunk 1]
```diff
@@ -5,19 +5,19 @@ import (
...
-func returnFast(rt *routing.Route) bool {
-	if rt.BackendType != eskip.LBBackend {
-		return true
-	}
+type fadeIn struct {
+	rnd              *rand.Rand
+	endpointRegistry *routing.EndpointRegistry
+}
 
+func (f *fadeIn) noFadeIn(rt *routing.Route) bool {
```
Comment (AlexanderYastrebov) [inline proxy/fadein.go:16]: Do we need this? Maybe its better to simply inline the condition

File: proxy/fadein.go [hunk 2]
```diff
@@ -26,18 +26,18 @@ func fadeIn(lifetime time.Duration, duration time.Duration, exponent float64) fl
...
 	now := time.Now()
-	threshold := rnd.Float64()
+	threshold := f.rnd.Float64()
 
 	filtered := make([]routing.LBEndpoint, 0, len(endpoints))
 	for _, e := range endpoints {
-		f := fadeIn(
-			now.Sub(registry.GetMetrics(e.Host).DetectedTime()),
+		f := f.fadeInScore(
+			now.Sub(f.endpointRegistry.GetMetrics(e.Host).DetectedTime()),
```
Comment (AlexanderYastrebov) [inline proxy/fadein.go:40]: Maybe extract this to a locak variable `age := now.Sub(f.endpointRegistry.GetMetrics(e.Host).DetectedTime())`

File: proxy/fadein.go [hunk 3]
```diff
@@ -5,19 +5,19 @@ import (
 	"math/rand"
 	"time"
 
-	"github.com/zalando/skipper/eskip"
 	"github.com/zalando/skipper/routing"
 )
 
-func returnFast(rt *routing.Route) bool {
-	if rt.BackendType != eskip.LBBackend {
```
Comment (MustafaSaber) [inline proxy/fadein.go:13]: Why did you remove this condition?

Comment (RomanZavodskikh) [reply proxy/fadein.go:13]: The filterFadeIn function is called only by `selectEndpoint` from proxy.go. However, the `selectEndpoint` is called only for `eskip.LBBackend`, see https://github.com/zalando/skipper/blob/v0.19.32/proxy/proxy.go#L505 .

This fact means there is no sense to check BackendType one more time, is was already checked before.

File: proxy/fadein.go (hunk 1 above)
Comment (RomanZavodskikh) [reply proxy/fadein.go]: It was intended to increase readability (literally tell the people reading this code "this is the case when there is no fade-in"), however, I agree it may be too obvious so I decided to inline it.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by RomanZavodskikh** [comment]
:+1:
